kernel/aster-nix/src/fs/fs_resolver.rs

```diff
@@ -411,3 +434,15 @@ pub fn split_path(path: &str) -> (&str, &str) {
 
     (dir_path, file_name)
 }
+
+bitflags! {
+    pub struct LookupFlags: u32 {
+        // pathwalk mode
+        const LOOKUP_FOLLOW		= 0x0001;	// Follow links at the end.
+        const LOOKUP_DIRECTORY	= 0x0002;	// Require a directory.
+        const LOOKUP_AUTOMOUNT	= 0x0004;   // Force terminal automount.
```

Comment:
When writing code for the `aster-nix` crate, you should think in a two-layered mental model: the top layer interfaces with the C-style, Linux public ABI, while the bottom layer provides the Rust-style, safe, internal API. We want to keep a clear boundary between the top and bottom layer and minimize the top layer because we absolutely do not want to work in the C style. In addition, one day we decide to implement another OS ABI (e.g., Windows NT) on top of the bottom layer.

Now back onto the proposed changes to this file. This file belongs to the bottom layer. So it should not define types such as `LookupFlags` in the C or Linux style.